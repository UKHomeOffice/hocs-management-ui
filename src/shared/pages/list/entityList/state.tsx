import EntityListItem from '../../../models/entityListItem';

export interface State {
    entities: EntityListItem[];
    entitiesLoaded: boolean;
}
