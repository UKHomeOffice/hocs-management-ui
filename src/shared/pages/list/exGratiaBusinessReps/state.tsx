import EntityListItem from '../../../models/entityListItem';

export interface State {
    representatives: EntityListItem[];
    representativesLoaded: boolean;
}
