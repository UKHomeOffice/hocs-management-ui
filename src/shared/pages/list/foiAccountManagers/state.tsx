import EntityListItem from '../../../models/entityListItem';

export interface State {
    accountManagers: EntityListItem[];
    accountManagersLoaded: boolean;
}
