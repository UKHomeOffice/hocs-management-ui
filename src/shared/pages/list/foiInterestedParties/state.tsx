import EntityListItem from '../../../models/entityListItem';

export interface State {
    parties: EntityListItem[];
    partiesLoaded: boolean;
}
