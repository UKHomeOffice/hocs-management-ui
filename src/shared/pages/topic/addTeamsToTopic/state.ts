export interface State {
    errorDescription: string;
    errorTitle: string;
    topicName: string;
    privateMinisterTeam: string;
    draftQATeam: string;
}
