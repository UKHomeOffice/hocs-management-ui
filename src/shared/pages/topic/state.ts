import Item from '../../models/item';

export interface State {
    displayName: string;
    selectedParentTopic?: Item;
}
