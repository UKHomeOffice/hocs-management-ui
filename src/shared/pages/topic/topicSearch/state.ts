export interface State {
    topicValue: string;
}
