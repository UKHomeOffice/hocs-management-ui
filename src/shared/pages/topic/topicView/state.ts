import Item from '../../../models/item';

export interface State {
    topic: Item;
    privateMinisterTeam?: Item;
    draftQATeam?: Item;
}
