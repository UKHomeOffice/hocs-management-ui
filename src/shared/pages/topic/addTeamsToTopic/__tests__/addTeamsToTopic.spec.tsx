import React from 'react';
import { createBrowserHistory, History, Location } from 'history';
import { act, wait, render, RenderResult, fireEvent, getByText } from '@testing-library/react';
import * as TeamsService from '../../../../services/teamsService';
import * as TopicsService from '../../../../services/topicsService';
import { State } from '../state';
import * as useError from '../../../../hooks/useError';
import { match, MemoryRouter } from 'react-router-dom';
import AddTeamsToTopicView from '../addTeamsToTopic';

let match: match<any>;
let history: History<any>;
let location: Location;
let mockState: State;

jest.mock('../../../../services/teamsService', () => ({
    __esModule: true,
    getTeam: jest.fn().mockReturnValue(Promise.resolve({
        label: '__label__',
        value: '__value__'
    }))
}));

jest.mock('../../../../services/topicsService', () => ({
    __esModule: true,
    getTopic: jest.fn().mockReturnValue(Promise.resolve({
        label: '__label__',
        value: '__value__'
    }))
}));

const getTeamSpy = jest.spyOn(TeamsService, 'getTeam');
const getTopicSpy = jest.spyOn(TopicsService, 'getTopic');
const useReducerSpy = jest.spyOn(React, 'useReducer');
const useErrorSpy = jest.spyOn(useError, 'default');

const renderComponent = () => render(
    <MemoryRouter>
        <AddTeamsToTopicView history={history} location={location} match={match}></AddTeamsToTopicView>
    </MemoryRouter>
);

beforeEach(() => {
    history = createBrowserHistory();
    location = {
        hash: '',
        key: '',
        pathname: '',
        search: '',
        state: {}
    };
    match = {
        isExact: true,
        params: { topicId: '__topicId__', privateMinister: '__privateMinister__', draftQA: '__draftQA__' },
        path: '',
        url: ''
    };
    mockState = {
        topic: {
            label: '__label__',
            value: '__value__'
        },
        privateMinisterTeam: {
            active: true,
            displayName: '__displayName__',
            letterName: '__letterName__',
            permissions: [],
            type: '__type__'
        },
        draftQaTeam: {
            active: true,
            displayName: '__displayName__',
            letterName: '__letterName__',
            permissions: [],
            type: '__type__'
        }
    };
    useReducerSpy.mockImplementationOnce(() => [mockState, jest.fn()]);
    useErrorSpy.mockImplementation(() => [{}, jest.fn(), jest.fn(), jest.fn()]);
});

describe('when the addTeamsToTopicView component is mounted', () => {
    it('should render with default props', async () => {
        expect.assertions(3);
        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await wait(() => {
            expect(getTeamSpy).toHaveBeenCalled();
            expect(getTopicSpy).toHaveBeenCalled();
            expect(wrapper.container).toMatchSnapshot();
        });
    });
});

describe('when the submit button is clicked', () => {
    it.skip('should push a new page into the history', async () => {
        history.push = jest.fn();
        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await wait(async () => {
            const submitButton = getByText(wrapper.container, 'Submit');
            fireEvent.click(submitButton);
        });

        expect(history.push).toHaveBeenCalledWith('/');
    });
});
