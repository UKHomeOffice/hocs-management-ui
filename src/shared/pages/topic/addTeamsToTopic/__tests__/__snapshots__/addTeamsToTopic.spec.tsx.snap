// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the addTeamsToTopicView component is mounted should render with default props 1`] = `
<div>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-two-thirds-from-desktop"
    >
      <a
        class="govuk-back-link"
        href="/topic/\${state.topicName}"
      >
        Back
      </a>
      <h1
        class="govuk-heading-xl"
      >
        Summary
      </h1>
      <h2
        class="govuk-heading-l"
      >
        Topic: __label__
      </h2>
    </div>
  </div>
  <div
    class="govuk-form-group"
  >
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-one-half-from-desktop"
      >
        <form
          method="POST"
        >
          <input
            name="_csrf"
            type="hidden"
            value=""
          />
          <table
            class="govuk-table"
          >
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Stages
                </th>
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Team assignment
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header"
                  scope="row"
                >
                  Draft/QA
                </th>
                <td
                  class="govuk-table__cell"
                >
                  __displayName__
                </td>
              </tr>
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header"
                  scope="row"
                >
                  Private Office/Minister
                </th>
                <td
                  class="govuk-table__cell"
                >
                  __displayName__
                </td>
              </tr>
            </tbody>
          </table>
          <input
            class="govuk-button"
            type="submit"
            value="Submit"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
